components/txn_types/src/write.rs

```diff
@@ -217,6 +277,10 @@ impl WriteRef<'_> {
         if self.has_overlapped_rollback {
             b.push(FLAG_OVERLAPPED_ROLLBACK);
         }
+        if let Some(ts) = self.gc_fence {
```

Comment:
Maybe we can calculate a better capacity estimation like https://github.com/tikv/tikv/blob/6b4466ba8a/components/txn_types/src/lock.rs#L161